azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -50,47 +50,46 @@ def gen_transport_cert(self, prv_file, crt_file):
 
     def get_pubkey_from_prv(self, file_name):
         if not os.path.exists(file_name):
-            raise IOError("File not found: {0}", file_name)
-
-        cmd = "{0} rsa -in {1} -pubout 2>/dev/null".format(self.openssl_cmd, 
-                                                           file_name)
-        pub = shellutil.run_get_output(cmd)[1]
-        return pub
+            raise IOError("File not found: {0}".format(file_name))
```

Comment:
Not the right way to raise IOError. While this sets the message, it doesn't set the errno member of the exception instance, it doesn't set the filename, and it puts too much information in strerror.

If you have to do this, this is the right way to do it:
```
raise IOError(errno.ENOENT, "File not found", file_name)
```